pkg/core/alias.go

```diff
@@ -22,19 +22,23 @@ var (
 	Now                   = time.Now
 	TempDir               = os.TempDir
 
-	SetupSignalHandler = func() context.Context {
+	SetupSignalHandler = func() (context.Context, context.Context) {
+		gracefulCtx, gracefulCancel := context.WithCancel(context.Background())
 		ctx, cancel := context.WithCancel(context.Background())
 		c := make(chan os.Signal, 2)
```

Comment:
Might want to make this 
```suggestion
		c := make(chan os.Signal, 3)
```

It's not clear to me what happens if the channel blocks on sending the third signal.

Reply:
I think it's unlikely that we will end up with 3 signals in the queue since we are just closing the contexts, but it's a good point anyway. Let me change it.